components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -138,33 +152,26 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 	}
 
 	err = Transact(db, func(tx *DBTrans) error {
+		logrus.Debugf("processing node %s with cloud info %s %s %s", node.GetName(), node.GetSourceId(), node.GetSourceAccountId(), node.GetSourceRegion())
+		nodeDetails := nodeDetails{
+			nodeState:           nodeState,
+			lastContact:         lastContact,
+			mgrType:             mgrType,
+			lastContactDataByte: lastContactDataByte,
+			projectsDataByte:    projectsDataByte,
+		}
+
 		if len(node.GetSourceId()) == 0 || len(node.GetSourceAccountId()) == 0 || len(node.GetSourceRegion()) == 0 {
-			if node.GetScanData() != nil {
-				_, err = db.Exec(sqlUpsertByIDScanData, node.GetUuid(),
-					node.GetName(), node.GetPlatformName(), node.GetPlatformRelease(),
-					nodeState, lastContact, node.GetSourceRegion(), node.GetSourceAccountId(),
-					node.GetJobUuid(), lastContactDataByte, projectsDataByte, mgrType)
-			} else if node.GetRunData() != nil {
-				_, err = db.Exec(sqlUpsertByIDRunData, node.GetUuid(),
-					node.GetName(), node.GetPlatformName(), node.GetPlatformRelease(),
-					nodeState, lastContact, node.GetSourceRegion(), node.GetSourceAccountId(),
-					lastContactDataByte, projectsDataByte, mgrType)
-			}
+			err = db.upsertByID(node, nodeDetails)
 		} else {
-			var id string
-			if node.GetScanData() != nil {
-				id, err = db.SelectStr(sqlUpsertBySourceIDScanData, node.GetUuid(),
-					node.GetName(), node.GetPlatformName(), node.GetPlatformRelease(),
-					nodeState, lastContact, node.GetSourceId(), node.GetSourceRegion(),
-					node.GetSourceAccountId(), node.GetJobUuid(), lastContactDataByte, projectsDataByte, mgrType)
-			} else if node.GetRunData() != nil {
-				id, err = db.SelectStr(sqlUpsertBySourceIDRunData, node.GetUuid(),
-					node.GetName(), node.GetPlatformName(), node.GetPlatformRelease(),
-					nodeState, lastContact, node.GetSourceId(), node.GetSourceRegion(),
-					node.GetSourceAccountId(), lastContactDataByte, projectsDataByte, mgrType)
-			}
-			if len(id) > 0 {
-				node.Uuid = id
+			node.Uuid, err = db.upsertByCloudDetails(node, nodeDetails)
```

Comment:
I'm a bit confused on why we are using `db` rather than `tx` here.  I think this implies that are happening outside of the transaction?

Reply:
hmm ya. good point, i missed that. ill see about switching to tx